src/Microsoft.ML/Runtime/Experiment/Experiment.cs

```diff
@@ -170,6 +170,11 @@ private string Serialize(string name, object input, object output)
             }
         }
 
+        public void AddSerialize(string name, object input, object output)
```

Comment:
Just wondering if this is the right name for the public API.  When a user extends ML.NET with a custom learner, they just want to Add it and don't care that internally we serialize, right?  Not sure what would be the best name though.

Reply:
Probably not. Not sure a user would notice it as this function is called from a piece of code automatically generated by a unit test. I would prefer to avoid Add as there are already many Add methods plus it does not just Add, the serialization forbids any further modifications to the added object. What do you think?

Reply:
Okay, that makes sense.  And sorry if this is self-evident, but what is a rough example of how this would look to the user adding a custom learner?

Reply:
You can have a look at this example: https://github.com/xadupre/machinelearningext/blob/master/machinelearningext/EntryPoints/EntryPointsFeaturesTransformsPolynom.cs.

Reply:
Nice example - I didn't realize you had created all those extensions.  I think AddSerialize makes sense.  Others I thought of were AddEntryPoint, AddNode, or AddToGraph in case you like any of them. (Are those all correct descriptions of what is happening?)

Reply:
Thanks. I like AddEntryPoint. I'll modify the PR.

Reply:
Thanks. I like AddEntryPoint. I'll modify the PR.